swap/types.go

```diff
@@ -49,3 +54,120 @@ type EmitChequeMsg struct {
 type ConfirmChequeMsg struct {
 	Cheque *Cheque
 }
+
+// Uint256 represents an unsigned integer of 256 bits
+type Uint256 struct {
+	value big.Int
+}
+
+var minUint256 = big.NewInt(0)
+var maxUint256 = new(big.Int).Sub(new(big.Int).Exp(big.NewInt(2), big.NewInt(256), nil), big.NewInt(1)) // 2^256 - 1
+
+// NewUint256 creates a Uint256 struct with a minimum initial underlying value
+func NewUint256() *Uint256 {
```

Comment:
I would expect to be able to set the new value right in the constructor, like `big.NewInt(val)`

Reply:
of course. i will explain my reasoning here: `NewUint256` returns _only_ a `Uint256` pointer, which is a comfortable condition to have. the same thing happens with the `Uint64ToUint256` call.

if we receive a `big.Int` param, we need to output an `err` variable as well, since it's possible that the received param falls outside of the allowed range (as it happens with `Set`).

this would not allow us to do calls like `newCumulativePayout, err := NewUint256().Add(cumulativePayout, castedPrice)` because the constructor now has 2 variables as output.

however, I am not opposed to adding a _new_ constructor with this behaviour. so instead of:
```
NewUint256().Set(*tentativeLiquidBalance)
```
we'd have something like:
```
NewUint256FromBigInt(*tentativeLiquidBalance)
```

although my choice for the `NewUint256FromBigInt` name is pretty bad.

(name suggestions/changes are welcome)